src/CommonDBVisible.php

```diff
@@ -434,8 +434,10 @@ public function showVisibility()
     protected function getShowVisibilityDropdownParams()
     {
         return [
-            'type'  => '__VALUE__',
-            'right' => strtolower($this::getType()) . '_public'
+            'type'          => '__VALUE__',
+            'right'         => strtolower($this::getType()) . '_public',
+            'entity'        => $this->fields['entities_id'],
+            'is_recursive'  => $this->fields['is_recursive'],
```

Comment:
`getShowVisibilityDropdownParams()` is also used on plugins. New parameters should only be added if there is a corresponding field.